### PR TITLE
add slack bot nodejs example

### PR DESCRIPTION
**Pull request by rylandg**
This is just the code review. I still need to update the README so it doesn't reccomend enabling the bot in all channels. I just figured it might be a lengthy review process and wanted to get it started.

This was tested in a dev-scratch Slack workspace extensively. Should be relatively safe due to `vm2`.

Here are some comments Michael had in slack (I've addressed them mostly)

```
1. trigger word should be env var
2. race is not good as the code continue to run. use https://nodejs.org/api/vm.html is you want to restrict run time
3. why embed NL when you can have a new line in backticked text? it will be more readable with real new line
4. finalMsg is really output message. should be const, by combining codeMessage with logMessage for example.
5. you only get console.log, err and others are lost.
6. regex is almost always wrong. yours is no exception. it matches, for example, a string of 8 backticks. also `\s\S` is `.`
7. console.log is overwritten with every call to the function
8. runcode does not just run the code, but also extract decode, run, format response and post it back. name the function properly. better yet, split it.
```

### CODE REVIEW

File: slack-code-runner-bot/function.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+ * @param {string} textToExtractFrom - segment to attempt code extraction on
+ *
+ * @return {string} - extracted code segment
+ */
+function extractBetweenBackticks(textToExtractFrom) {
+  const startIdx = textToExtractFrom.indexOf(triBacktick);
+  const endIdx = textToExtractFrom.lastIndexOf(triBacktick);
+
+  if (startIdx === -1 || endIdx === -1 || (startIdx === endIdx)) {
+    throw new Error('Javascript input must begin and end with triple backticks');
```
Comment (arielshaqed) [inline slack-code-runner-bot/function.js]: 1. Nit: "JavaScript".
  a. Meganit: "ECMAScript" :-/
2. The error message is unclear: the message doesn't have to start and end with triple backticks.  Maybe something like "No code to run found between triple backticks"?

File: slack-code-runner-bot/function.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  }
+  return extractText.substring(startIdx + 3, endIdx);
+}
```
Comment (arielshaqed) [inline slack-code-runner-bot/function.js]: Would this function be easier to write with a regexp ``/\`\`\`(.*)\`\`\`/``?  (*Please* say "yes", marking this regexp down was incredibly tricky...).

File: slack-code-runner-bot/function.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+/**
+ * Runs the provided code and returns computed output.
+ *
+ * @param {string} code - Javascript code to evaluate
+ * @param {Array} logBuffer - buffer to insert stdout ouput into
```
Comment (arielshaqed) [inline slack-code-runner-bot/function.js]: Nit: "output" (with a `T`).

File: slack-code-runner-bot/function.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    err: pushLogs,
+    log: pushLogs,
+    warn: pushLogs,
+  };
+
+  let result;
+  const vm = new VM({
+    timeout: maxTimeMS,
+    sandbox: { console: redirectConsole },
+    require: { external: true },
```
Comment (arielshaqed) [inline slack-code-runner-bot/function.js]: How safe is this?  *Any* external module?  (I can install what I like if I use `child_process` to run `npm`, or if I just download the code with `http` and require it).

File: slack-code-runner-bot/function.js (hunk 1 above)
Comment (rylandg) [reply slack-code-runner-bot/function.js]: 1. Will fix
2. I don't understand completely. If there is not two sets of ``` at the start and end of the text this case is hit. In fact I don't even handle the case of empty '``````'.

File: slack-code-runner-bot/function.js (hunk 2 above)
Comment (rylandg) [reply slack-code-runner-bot/function.js]: It's funny, I originally had it as regexp and Michael suggested I change it in slack. I should have posted those comments he made in this PR (I've done that now).

File: slack-code-runner-bot/function.js (hunk 3 above)
Comment (rylandg) [reply slack-code-runner-bot/function.js]: nice catch

File: slack-code-runner-bot/function.js (hunk 4 above)
Comment (rylandg) [reply slack-code-runner-bot/function.js]: I honestly am not 100% sure based on the docs. We can air on the safe side and initially just block `require`.

**Comment by rylandg** [comment]
@arielshaqed I've addressed your high level feedback. Will fix README today.

**Comment by rylandg** [comment]
@arielshaqed Consider it done.

**Comment by rylandg** [comment]
I've gone over the README one last time to assure myself that the demo is reproducible.

**Pull request merged**